core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -157,12 +155,12 @@ void SYCL::impl_initialize(InitializationSettings const& settings) {
     !defined(KOKKOS_ARCH_VOLTA) && !defined(KOKKOS_ARCH_TURING75) &&   \
     !defined(KOKKOS_ARCH_AMPERE)
   if (!settings.has_device_id() && gpu_devices.empty()) {
-    Impl::SYCLInternal::singleton().initialize(sycl::device());
+    Impl::SYCLInternal::singleton().initialize(sycl::device(), 0);
     return;
   }
 #endif
-  using Kokkos::Impl::get_gpu;
-  Impl::SYCLInternal::singleton().initialize(gpu_devices[get_gpu(settings)]);
+  const auto id = ::Kokkos::Impl::get_gpu(settings);
+  Impl::SYCLInternal::singleton().initialize(gpu_devices[id], id);
```

Comment:
No need to propagate this to `SYCLInternal::initialize`.  Do
```suggestion
  Impl::SYCLInternal::singleton().initialize(gpu_devices[id]);
  Impl::SYCLInternal::m_syclDev = id;
```